doc/logrotate/logrotate_hld.md

```diff
@@ -0,0 +1,251 @@
+<!-- omit in toc -->
+###### Figure 1: Log rotate system chart diagram
+
+This feature requires access to SONiC DB. All log rotate related configuration (disk percentage, frequency and size, etc...) saved into the SONiC config database.
+Hostcfgd will listen for the configuration changes in corresponding tables and restart logrotate-config service. Log rotate config service - it is simple SystemD service.
+It reads log rotate configuration from database and apply it to log rotate config files.
+
+
+**The log rotate config files will be used:**
+1. /etc/logrotate.d/rsyslog - generic log files
+2. /etc/logrotate.d/debug - debug log files
+3. /etc/logrotate.d/syslog - syslog files
+
+## 2.2 Flows
+<!-- omit in toc -->
+### 2.2.1 Log rotate init flow
+![Log rotate init flow](images/logrotate_init_diagram.png "Figure 2: Log rotate init flow")
+<!-- omit in toc -->
+###### Figure 2: Log rotate init flow
+
+<!-- omit in toc -->
+### 2.2.2 Log rotate config flow
+![Log rotate config flow](images/logrotate_config_diagram.png "Figure 3: Log rotate flow")
+<!-- omit in toc -->
+###### Figure 3: Log rotate config flow
+
+<!-- omit in toc -->
+### 2.2.3 Log rotate show configuration
+![Log rotate show configuration](images/logrotate_show_diagram.png "Figure 4: Log rotate configuration")
+<!-- omit in toc -->
+###### Figure 4: Log rotate show configuration
+
+
+## 2.3 CLI
+
+<!-- omit in toc -->
+### 2.3.1 Command structure
+
+**User interface**:
+```
+config
+\-- logrotate
+    \-- <syslog|debug>
+        |-- disk-percentage <disk-percentage>
+        |-- frequency <freqency>
+        |-- max-number <max-number>
+        |-- size <size>
+
+show
+\-- logrotate
+```
+
+**Options:**
+
+General:
+1. `<disk-precentage>` - Rotate logs when they surpass a specified percentage of disk: `float`
+2. `<freqency>`        - Log files rotation frequency: `(enum:daily, weekly, monthly, yearly | string)`
+3. `<max-number>`      - Max number of log files to keep: `integer`
+4. `<size>`            - Rotate logs if they grow bigger than size in Mebibytes: `float`
+
+
+#### 2.3.1 Config command group
+**The following command set log rotate disc percentage:**
+```bash
+config logrotate <syslog|debug> disk-percentage <disk-percentage>
+```
+
+**The following command set log rotate frequency:**
+```bash
+config logrotate <syslog|debug> frequency <freqency>
+```
+
+**The following command set log rotate max-number:**
+```bash
+config logrotate <syslog|debug> max-number <max-number>
+```
+
+**The following command set log rotate size:**
+```bash
+config logrotate <syslog|debug> size <size>
+```
+
+#### 2.3.2 Show command
+**The following command display log rotate configuration:**
+```bash
+root@sonic:/home/admin# show logrotate
+file      disk-percentage  frequency      max-number    size
+------  -----------------  -----------  ------------  ------
+syslog             10.2    daily                  10    20.0
+debug              50.5    weekly                 20    10.0
```

Comment:
For the size column, please make it clear that this is mebibytes.

Reply:
Make sense